File: Practice/07/C++/07.cpp
#include <iostream>
using namespace std;

int main()
{
	setlocale(LC_ALL, "Russian");
	double S;
	double a, b, c;
	int choice;
	cin >> choice;
	if (choice == 1)
	{
		cin >> a >> b >> c;
		if (a < b + c) && (b < a + c) && (c < a + b)
		{
			double p = (a + b + c) / 2;
			S = sqrt(p * (p - a) * (p - b) * (p - c));
			cout << "S = " << S;
		}
		else cout << "Такого треугольника не существует.";
	}
	else if (choice == 2)
	{
		double Ax, Ay, Bx, By, Cx, Cy;
		cin >> Ax >> Ay;
		cin >> Bx >> By;
		cin >> Cx >> Cy;
		a = sqrt((Bx - Ax) * (Bx - Ax) + (By - Ay) * (By - Ay));
		b = sqrt((Cx - Bx) * (Cx - Bx) + (Cy - By) * (Cy - By));
		c = sqrt((Cx - Ax) * (Cx - Ax) + (Cy - Ay) * (Cy - Ay));
		if (a < b + c) && (b < a + c) && (c < a + b)
		{
			double p = (a + b + c) / 2;
			S = sqrt(p * (p - a) * (p - b) * (p - c));
			cout << "S = " << S;
		}
		else cout << "Такого треугольника не существует.";
	}
	else cout << "Ошибочный ввод.";
}

File: Practice/07/Python/07.py
import math

choice = int(input())
if choice == 1:
    a = float(input())
    b = float(input())
    c = float(input())
    if (a < b + c) & (b < a + c) & (c < a + b):
        p = (a + b + c) / 2
        S = math.sqrt(p * (p - a) * (p - b) * (p - c))
        print("S =", S)
    else: print("Такого треугольника не существует.")
elif choice == 2:
    Ax, Ay = map(float, input().split())
    Bx, By = map(float, input().split())
    Cx, Cy = map(float, input().split())
    a = math.sqrt((Bx - Ax) * (Bx - Ax) + (By - Ay) * (By - Ay))
    b = math.sqrt((Cx - Bx) * (Cx - Bx) + (Cy - By) * (Cy - By))
    c = math.sqrt((Cx - Ax) * (Cx - Ax) + (Cy - Ay) * (Cy - Ay))
    if (a < b + c) & (b < a + c) & (c < a + b):
        p = (a + b + c) / 2
        S = math.sqrt(p * (p - a) * (p - b) * (p - c))
        print("S =", S)
    else: print("Такого треугольника не существует.")
else:
    print("Ошибочный ввод")
